synapse/events/snapshot.py

```diff
@@ -13,56 +13,54 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import abc
+
 from frozendict import frozendict
 
 from twisted.internet import defer
 
 
-class EventContext(object):
+class StatelessContext(object):
```

Comment:
this feels like a funny name, given it may have a state.

Indeed I wonder if having it separate from  `EventContext` is worthwhile? maybe combine them and have `DeserializedContext` just override the accessors?

Reply:
Unfortunately we can't just override accessors since they would need to return a deferreds in the DeserializedContext

Reply:
yes, but that's also true for `EventContext`. I don't understand why that means we can't override the accessors.